File: 0739-daily-temperatures/0739-daily-temperatures.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        Stack<Integer> stack = new Stack<>(); // incies
        int[] result = new int[temperatures.length];
        for(int i=0; i<temperatures.length; i++){
            while(!stack.isEmpty() && temperatures[i]> temperatures[stack.peek()]){
                int prevs = stack.pop();
                result[prevs] = i- prevs;
            }
            stack.push(i);
        }
        return result;
    }
}

File: 0739-daily-temperatures/0739-daily-temperatures.py
class Solution(object):
    def dailyTemperatures(self, temperatures):
        """
        :type temperatures: List[int]
        :rtype: List[int]
        """
        n = len(temperatures) 
        stack = []
        result = [0]*n

        for i, temp in enumerate(temperatures):

            while stack and temp > temperatures[stack[-1]]:
                previous_index = stack.pop()
                result[previous_index] = i - previous_index

            stack.append(i)

        return result